service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java

```diff
@@ -68,7 +78,17 @@ public ReferencedResource createReferenceResourceForClone(
       ReferencedResource resourceToClone,
       ReferencedResource sourceReferencedResource,
       AuthenticatedUserRequest userRequest) {
+
```

Comment:
should we do this check for all cloning endpoint? I know the UI don't let you clone a workspace partially right now but the wsm service should handle that flow correctly regarding policy.

Reply:
We do need the check for all clone paths. Any clone needs to validate and merge, not only the reference case; the copy controlled resource definition and copy controlled resource data cases, too.